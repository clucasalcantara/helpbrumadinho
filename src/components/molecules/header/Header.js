import React from 'react'
import { array } from 'prop-types'
import { Link } from 'react-router-dom'
// Styles
import './style.scss'

const Header = ({ navItems }) => {
  return (
    <header className="fixed-top hb-header">
      <div className="ui container">
        <a href="/" className="hb-logo">
          Help Brumadinho
        </a>
        <button type="button" className="menu-toggle" id="btnMenu">
          <div className="one" />
          <div className="two" />
          <div className="three" />
        </button>
        <nav className="hb-menu">
          {navItems.map(item => (
            <Link key={item.path} to={item.path} className="item">
              {item.text}
            </Link>
          ))}
        </nav>
      </div>
    </header>
  )
}

Header.propTypes = {
  navItems: array
}

Header.defaultProps = {
  navItems: []
}

export default Header
