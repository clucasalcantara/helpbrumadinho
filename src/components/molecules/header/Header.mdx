---
name: Header
menu: Molecules
---

import { Playground, PropsTable } from 'docz'
import Header from './Header'

# Header component

## Basic usage

<Playground>
  <Header />
</Playground>

## Properties

<PropsTable of={Header} />
