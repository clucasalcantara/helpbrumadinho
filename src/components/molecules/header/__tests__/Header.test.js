import React from 'react'
import { shallow } from 'enzyme'
import Header from '../Header'

describe('Header component', () => {
  it('should renders without crash', () => {
    const wrapper = shallow(
      <Header navItems={[{ text: 'Work with coffee', path: '/' }]} />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
