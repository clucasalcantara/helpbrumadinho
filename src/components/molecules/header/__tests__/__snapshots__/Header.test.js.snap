// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should renders without crash 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    navItems={
      Array [
        Object {
          "path": "/",
          "text": "Work with coffee",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="ui container"
      >
        <a
          className="hb-logo"
          href="/"
        >
          Help Brumadinho
        </a>
        <button
          className="menu-toggle"
          id="btnMenu"
          type="button"
        >
          <div
            className="one"
          />
          <div
            className="two"
          />
          <div
            className="three"
          />
        </button>
        <nav
          className="hb-menu"
        >
          <Link
            className="item"
            replace={false}
            to="/"
          >
            Work with coffee
          </Link>
        </nav>
      </div>,
      "className": "fixed-top hb-header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            className="hb-logo"
            href="/"
          >
            Help Brumadinho
          </a>,
          <button
            className="menu-toggle"
            id="btnMenu"
            type="button"
          >
            <div
              className="one"
            />
            <div
              className="two"
            />
            <div
              className="three"
            />
          </button>,
          <nav
            className="hb-menu"
          >
            <Link
              className="item"
              replace={false}
              to="/"
            >
              Work with coffee
            </Link>
          </nav>,
        ],
        "className": "ui container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Help Brumadinho",
            "className": "hb-logo",
            "href": "/",
          },
          "ref": null,
          "rendered": "Help Brumadinho",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="one"
              />,
              <div
                className="two"
              />,
              <div
                className="three"
              />,
            ],
            "className": "menu-toggle",
            "id": "btnMenu",
            "type": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "one",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "two",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "three",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="item"
                replace={false}
                to="/"
              >
                Work with coffee
              </Link>,
            ],
            "className": "hb-menu",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "/",
              "nodeType": "class",
              "props": Object {
                "children": "Work with coffee",
                "className": "item",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Work with coffee",
              "type": [Function],
            },
          ],
          "type": "nav",
        },
      ],
      "type": "div",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="ui container"
        >
          <a
            className="hb-logo"
            href="/"
          >
            Help Brumadinho
          </a>
          <button
            className="menu-toggle"
            id="btnMenu"
            type="button"
          >
            <div
              className="one"
            />
            <div
              className="two"
            />
            <div
              className="three"
            />
          </button>
          <nav
            className="hb-menu"
          >
            <Link
              className="item"
              replace={false}
              to="/"
            >
              Work with coffee
            </Link>
          </nav>
        </div>,
        "className": "fixed-top hb-header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className="hb-logo"
              href="/"
            >
              Help Brumadinho
            </a>,
            <button
              className="menu-toggle"
              id="btnMenu"
              type="button"
            >
              <div
                className="one"
              />
              <div
                className="two"
              />
              <div
                className="three"
              />
            </button>,
            <nav
              className="hb-menu"
            >
              <Link
                className="item"
                replace={false}
                to="/"
              >
                Work with coffee
              </Link>
            </nav>,
          ],
          "className": "ui container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Help Brumadinho",
              "className": "hb-logo",
              "href": "/",
            },
            "ref": null,
            "rendered": "Help Brumadinho",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="one"
                />,
                <div
                  className="two"
                />,
                <div
                  className="three"
                />,
              ],
              "className": "menu-toggle",
              "id": "btnMenu",
              "type": "button",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "one",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "two",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "three",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  className="item"
                  replace={false}
                  to="/"
                >
                  Work with coffee
                </Link>,
              ],
              "className": "hb-menu",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "/",
                "nodeType": "class",
                "props": Object {
                  "children": "Work with coffee",
                  "className": "item",
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Work with coffee",
                "type": [Function],
              },
            ],
            "type": "nav",
          },
        ],
        "type": "div",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
